packages/core/helper-plugin/lib/src/components/GenericInput/index.js

```diff
@@ -166,7 +166,7 @@ const GenericInput = ({
           hint={hint}
           name={name}
           onChange={date => {
-            const formattedDate = date.toISOString();
+            const formattedDate = date.toISOString().split('T')[0];
```

Comment:
How about this instead?

```js
const tzAdjustedDate = new Date(date.getTime() - date.getTimezoneOffset() * 60000);
const formattedDate = tzAdjustedDate.toISOString().slice(0,10);
```

Reply:
Unfortunately, this is not enough. On the display side, we create a Date. We have to change that too.